src/librustc_driver/lib.rs

```diff
@@ -542,16 +541,18 @@ fn handle_explain(registry: Registry, code: &str, output: ErrorOutputType) {
                 }
                 text.push('\n');
             }
-
             if stdout_isatty() {
                 show_content_with_pager(&text);
             } else {
                 print!("{}", text);
             }
         }
-        None => {
+        Ok(None) => {
             early_error(output, &format!("no extended information for {}", code));
         }
+        Err(_invalid_error_code) => {
```

Comment:
Sorry, I literally meant the InvalidErrorCode struct from the other file (you may need to import it).